compiler/noirc_evaluator/src/ssa/opt/fill_internal_slices.rs

```diff
@@ -0,0 +1,705 @@
+//! This module defines the internal slices data fill pass.
+//! The purpose of this pass is to fill out nested slice values represented by SSA array values.
+//! "Filling out" a nested slice specifically refers to making a nested slice's internal slice types
+//! match up in their size. This pass is necessary for dynamic array operations to work in ACIR gen
+//! as we need to have a known size for any memory operations. As slice types do not carry a size we
+//! need to make sure all nested internal slices have the same size in order to accurately
+//! read from or write to a nested slice. This pass ultimately attaches dummy data to any smaller internal slice types.
+//!
+//! A simple example:
+//! If we have a slice of the type [[Field]] which is of length 2. The internal slices themselves
+//! could be of different sizes, such as 3 and 4. An array operation on this nested slice would look
+//! something like below:
+//! array_get [[Field 3, [Field 1, Field 1, Field 1]], [Field 4, [Field 2, Field 2, Field 2, Field 2]]], index Field v0
+//! Will get translated into a new instruction like such:
+//! array_get [[Field 3, [Field 1, Field 1, Field 1, Field 0]], [Field 4, [Field 2, Field 2, Field 2, Field 2]]], index Field v0
```

Comment:
Shouldn't this be:
```suggestion
//! array_get [Field 3, [Field 1, Field 1, Field 1], Field 4, [Field 2, Field 2, Field 2, Field 2]], index Field v0
//! Will get translated into a new instruction like such:
//! array_get [Field 3, [Field 1, Field 1, Field 1, Field 0], Field 4, [Field 2, Field 2, Field 2, Field 2]], index Field v0
```

Reply:
Yup! Don't know where I got the extra brackets from as I ripped this example from the unit test below which does not have the brackets.